File: golang/62.unique-paths.go
/*
 * @lc app=leetcode id=62 lang=golang
 *
 * [62] Unique Paths
 */
package main

// @lc code=start
func uniquePaths(m int, n int) int {
	return C(m+n, m)
	// dp := make([][]int, m)
	// for i := 0; i < m; i++ {
	// 	dp[i] = make([]int, n)
	// }
	// for i := 0; i < m; i++ {
	// 	dp[i][0] = 1
	// }
	// for i := 0; i < n; i++ {
	// 	dp[0][i] = 1
	// }
	// for i := 0; i < m; i++ {
	// 	for j := 0; j < n; j++ {
	// 		// if i == 0 && j == 0 {
	// 		// 	dp[i][j] = 1
	// 		// } else if i == 0 {
	// 		// 	dp[i][j] = dp[i][j-1]
	// 		// } else if j == 0 {
	// 		// 	dp[i][j] = dp[i-1][j]
	// 		// } else {
	// 		dp[i][j] = dp[i-1][j] + dp[j-1][i]
	// 		// }
	// 	}
	// }
	// return dp[m-1][n-1]
}

// @lc code=end

File: cpp/62.unique-paths.cpp
/*
 * @lc app=leetcode id=62 lang=cpp
 *
 * [62] Unique Paths
 */
#include<leetcode-helper.h>

// @lc code=start
class Solution {
public:
    int uniquePaths(int m, int n) {
        int dp[m][n];
        for (auto i = 0; i < m;i++){
            for (auto j = 0; j < n;j++){
                if (i==0||j==0){
                    dp[i][j] = 1;
                }else{
                    dp[i][j] = 0;
                }
            }
        }
        for (auto i = 1; i < m;i++){
            for (auto j = 1; j < n;j++){
                dp[i][j] = dp[i][j - 1] + dp[i - 1][j];
            }
        }
        return dp[m - 1][n - 1];
    }
};
// @lc code=end

